syntax = "proto3";

package crm;

service CustomerService {
    rpc CreateCustomer (CreateCustomerRequest) returns (CustomerResponse);
    rpc GetCustomer (GetCustomerRequest) returns (CustomerResponse);
    rpc UpdateCustomer (UpdateCustomerRequest) returns (CustomerResponse);
    rpc DeleteCustomer (DeleteCustomerRequest) returns (DeleteCustomerResponse);
    rpc ListCustomers (ListCustomersRequest) returns (ListCustomersResponse);
}

service OrderService {
    rpc CreateOrder (CreateOrderRequest) returns (OrderResponse);
    rpc GetCustomerOrder (GetCustomerOrderRequest) returns (OrderResponse);
}

message CreateCustomerRequest {
    string name = 1;
    string email = 2;
}

message GetCustomerRequest {
    string id = 1;
}

message UpdateCustomerRequest {
    string id = 1;
    string name = 2;
    string email = 3;
}

message DeleteCustomerRequest {
    string id = 1;
}

message DeleteCustomerResponse {
    bool success = 1;
}

message ListCustomersRequest {
    int32 page = 1;
    int32 limit = 2;
}

message CustomerResponse {
    string id = 1;
    string name = 2;
    string email = 3;
    string created_at = 4;
}

message ListCustomersResponse {
    repeated CustomerResponse customers = 1;
    int32 total = 2;
}

message CreateOrderRequest {
    string customer_id = 1;
    string product_name = 2;
    double price = 3;
}

message GetCustomerOrderRequest {
    string customer_id = 1;
}

message OrderResponse {
    string id = 1;
    string customer_id = 2;
    string product_name = 3;
    double price = 4;
    string created_at = 5;
}